tests/test_filer.py

```diff
@@ -199,11 +205,172 @@ class FilerTest_no_env(unittest.TestCase, AssertThrowsMixin):
+        self.assertEqual(self.trans.get_names(), ('mybucket', '/folder/file.txt'))
+
+
+    def test_check_if_bucket_exists_success(self):
+        """
+        Check if the bucket exists
+        """
+        client = boto3.resource('s3', endpoint_url=None)
+        self.assertEqual(self.trans.check_if_bucket_exists(client),0)
+
+    def test_check_if_bucket_exists_error(self):
+        """
+        Check if bucket does not exist
+        """
+        path = "/home/user/filer_test/"
+        url = "http://s3.localhost:9000/mybucket/test/bucket_file1.txt"
+        ftype = "FILE"
+        trans = S3Transput(path, url, ftype)
+        client = boto3.resource('s3', endpoint_url=None)
+        self.assertEqual(trans.check_if_bucket_exists(client),1)
+
+    @patch("tesk_core.filer.os.path.exists", return_value=1)
+    def test_extraction_point_file(self,mock_path_exists):
+        """
+        Testing extraction of endpoint url from default file location
+        """
+        read_data = '\n'.join(["[default]", "endpoint_url = http://s3.localhost:9000"])
+        with patch("builtins.open", mock_open(read_data=read_data), create=True) as mock_file:
+            mock_file.return_value.__iter__.return_value = read_data.splitlines()
+            self.assertEqual(self.trans.extract_endpoint(),"http://s3.localhost:9000")
+
+    @patch.dict(os.environ, {"AWS_CONFIG_FILE": "~/.aws/config"})
+    def test_extraction_point_environ(self):
+        """
+        Testing successful extraction of endpoint url read from file path saved on enviornment variable
+        """
+        read_data = '\n'.join(["[default]","endpoint_url = http://s3.localhost:9000"])
+        with patch("builtins.open", mock_open(read_data=read_data),create=True) as mock_file:
+            mock_file.return_value.__iter__.return_value = read_data.splitlines()
+
+            self.assertEqual(self.trans.extract_endpoint(),"http://s3.localhost:9000")
+
+
+    @patch("tesk_core.filer.os.path.exists", return_value=0)
+    def test_extraction_point_error(self,mock_path_exists):
+        """
+        Checking if default file is not found
+        """
+        with self.assertRaises(FileNotFoundError):
+            self.trans.extract_endpoint()
+
+
+    @patch('tesk_core.filer.os.makedirs')
+    @patch("tesk_core.filer.S3Transput.get_names", return_value=("tesk", "test/"))
+    @patch("tesk_core.filer.S3Transput.get_s3_file", return_value=0)
+    @patch("tesk_core.filer.S3Transput.extract_endpoint", return_value=None)
+    def test_download_file_success(self,mock_extract_endpoint, mock_get_s3_file, mock_get_names, mock_folder_create ):
+        """
+        Checking for successful file download from Object storage server
+        """
+        self.trans.bucket_obj = self.conn.Bucket(self.trans.bucket)
+        self.conn.Bucket('tesk').put_object(Bucket='tesk', Key='test/bucket_file1.txt', Body='')
+        self.assertEqual(self.trans.download_file(), 0)
+
```

Comment:
Why is this `@patch("tesk_core.filer.S3Transput.get_names", return_value=("tesk", "test/"))` needed?
The test `test_get_names` suggests that `get_names` actually returns a name of the bucket and the full path to the file. Here, it looks like you expect a path to a directory and you need to mock it. Why?
Also why is extracting endpoint mocked here, if it is not really being called in the test?

Reply:
Also - we discussed it already - is there a chance to put S3Transput to a separate file?

Reply:
Removed the unnecessary mocks and moved the S3Transput class to a separate file. 

Note: When S3Transput was referenced(imported) from a new file it introduced cyclic import error so refactored the code by moving the **Transput** and **Type** class to filer_class.py.